bin/run-tests.sh

```diff
@@ -31,12 +33,12 @@ for test_dir in "${tmp_dir}"/*; do
 
     bin/run.sh "${test_dir_name}" "${test_dir_path}" "${test_dir_path}"
 
-    for file in "$results_file_path" "$expected_results_file_path"; do
-        # We sort both the '.message' values in results.json and expected_results.json files
-        tmp_file=$(mktemp -p "$test_dir/")
-        sorted_message=$(cat $file | jq -r '.message' >"$tmp_file" && sort "$tmp_file")
-        jq --arg msg "$sorted_message" '.message = $msg' "$file" >"$tmp_file" && mv "$tmp_file" "$file"
-    done
+    has_message=$(jq 'has("message") and .message != null' "$results_file_path")
+
+    if [ "$has_message" = "true" ]; then
+        sorted_message=$(jq -r '.message' "$results_file_path" | sed 's/^ *//' | "$SORTBIN" | jq -s -R '.')
+        jq ".message = $sorted_message" "$results_file_path" >"$results_file_path.tmp" && mv "$results_file_path.tmp" "$results_file_path"
```

Comment:
It's a bit strange to be testing only this command's exit status.

Prefer using `jq --arg` or `--argjson` to pass data into JQ.


```suggestion
        jq --arg message "$sorted_message" '.message = $message' "$results_file_path" >"$results_file_path.tmp"
        mv "$results_file_path.tmp" "$results_file_path"
```

Reply:
The `&&` means the `mv` only runs if `jq` has exit code 0.

Is the exit code of this `jq` being checked for any particular reason? This script otherwise doesn't check very many exit codes. It stands out to me that this particular exit code is being checked when almost no other exit codes are being checked.

Checking the exit code is entirely fine. But if you're doing so on this line, it's worth asking about the other lines ... and what behavior you want if `jq` has a non-zero exit code.